pkg/test/kube/accessor.go

```diff
@@ -389,13 +389,19 @@ func (a *Accessor) CreateNamespace(ns string, istioTestingAnnotation string) err
 	return err
 }
 
-func (a *Accessor) CreateNamespaceWithInjectionEnabled(ns string, istioTestingAnnotation string, configNamespace string) error {
+// CreateNamespaceWithInjectionEnabled with the given name and have sidecar-injection enabled.
+func (a *Accessor) CreateNamespaceWithInjectionEnabled(ns string, istioTestingAnnotation string,
+	configNamespace string, useCustomSidecarInjector bool) error {
```

Comment:
The `configNamespace` is only used when `useCustomSidecarInjector=true`.  I suggest for this method, merging these into a single argument `customSidecarInjectorNamespace`. If it's `""` then the label will not be set.

Reply:
@nmittler I agree with your point. your suggestion is working from this method view. but `ConfigNamespace` is a namespace in which config, discovery and auto-injector are deployed. it is always not empty for the new installer.

Reply:
The arguments serve the method ... the callers should set the arguments appropriately based on what the method needs.